psycopg/psycopg/generators.py

```diff
@@ -149,6 +149,63 @@ def fetch(pgconn: PGconn) -> PQGen[Optional[PGresult]]:
     return pgconn.get_result()
 
 
+def pipeline_communicate(pgconn: PGconn) -> PQGen[List[List[PGresult]]]:
+    """Generator to send queries from a connection in pipeline mode while also
+    receiving results.
+
+    Return a list results, including single PIPELINE_SYNC elements.
+
+    We wait for the connection socket to be read-ready or write-ready.
+    If read-ready, the server output buffer will be consumed completely until
+    no more results are available. Then, if write-ready, the client buffer
+    will be flushed the function returns.
+
+    When the client buffer got flushed but the socket is not read-ready,
+    meaning we cannot get results from the server, this will return no
+    results. In that case, the caller should issue pgconn.send_flush_request()
+    or pgconn.pipeline_sync() and retry, or wait for the server to flush its
+    output buffer.
+    """
+    results = []
+
+    while True:
+        ready = yield Wait.RW
+
+        if ready & Ready.R:
+            # This requires that the server has its output buffer flushed,
+            # which happens either automatically upon PQpipelineSync() or by
+            # calling PQsendFlushRequest().
+            pgconn.consume_input()
+
+            # Consume notifies
+            while 1:
```

Comment:
I guess you copied this from my code. `while 1` is a pattern I can't take off my muscle memory...

Reply:
Yes :) /me trying to be consistent with existing style... I wouldn't mind if we'd change all these!